test/bytecode.ts

```diff
@@ -19,3 +19,52 @@ export async function builtAndDeployedMetadataCoincide(
 
   return metadata(code) === metadata(contractArtifacts.deployedBytecode);
 }
+
+export async function readImmutables(
+  fullyQualifiedContractName: string,
+  contractAddress: string,
+): Promise<string[]> {
```

Comment:
Nice function!

I'd propose not to use this function directly, but create a wrapper `readSettlementImmutables` that calls this function and nicely formats the output into an object of the form `{ authenticatior, vault }` that can be destructured in the tests.
You could also only use `immutableAsAddress` in this wrapper.